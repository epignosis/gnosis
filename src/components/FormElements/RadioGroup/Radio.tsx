import React, { FC, InputHTMLAttributes } from "react";
import { SerializedStyles } from "@emotion/react";
import { InputSize } from "../Input/Input";
import { radioButtonContainer } from "./styles";
import { ExtendableProps } from "types/common";

export type RadioOption = {
  value: string;
  label: string;
  disabled?: boolean;
};

export type RadioProps = ExtendableProps<
  InputHTMLAttributes<HTMLInputElement>,
  RadioOption & {
    id: string;
    size?: InputSize;
    inline?: boolean;
  }
>;

const Radio: FC<RadioProps> = (props) => {
  const { id, label, size = "md", inline = false, ...rest } = props;

  return (
    <div css={(theme): SerializedStyles => radioButtonContainer(theme, { size, inline })}>
      <input id={id} type="radio" {...rest} />
      <label htmlFor={id}>{label}</label>
    </div>
  );
};

export default Radio;
