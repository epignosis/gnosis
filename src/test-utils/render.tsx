import React, { FC, ReactElement, ComponentType } from "react";
import { render, RenderResult, RenderOptions } from "@testing-library/react";
import ThemeProvider from "../components/ThemeProvider/ThemeProvider";

const AllTheProviders: FC = ({ children }) => <ThemeProvider>{children}</ThemeProvider>;

const customRender = (ui: ReactElement, options: RenderOptions): RenderResult =>
  render(ui, { wrapper: AllTheProviders as ComponentType, ...options });

export * from "@testing-library/react";

export { customRender as render };
