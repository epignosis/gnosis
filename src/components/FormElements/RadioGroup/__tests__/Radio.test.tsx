import React from "react";
import userEvent from "@testing-library/user-event";
import { faker } from "@faker-js/faker";
import Radio from "../Radio";
import { screen, render } from "@test-utils/render";

describe("<Radio />", () => {
  it("renders properly", () => {
    const labelTxt = faker.random.words();
    const value = faker.random.alphaNumeric();
    render(<Radio id={value} label={labelTxt} value={value} />);

    const radio = screen.getByLabelText(labelTxt);
    const label = screen.getByText(labelTxt);

    expect(radio).toBeInTheDocument();
    expect(label).toHaveTextContent(labelTxt);
  });

  it("renders disabled", () => {
    const labelTxt = faker.random.words();
    const value = faker.random.alphaNumeric();
    render(<Radio id={value} label={labelTxt} value={value} disabled />);

    const radio = screen.getByLabelText(labelTxt);

    expect(radio).toBeDisabled();
  });

  it("gets checked", () => {
    const labelTxt = faker.random.words();
    const value = faker.random.alphaNumeric();
    render(<Radio id={value} label={labelTxt} value={value} />);

    const radio = screen.getByLabelText(labelTxt);

    expect(radio).not.toBeChecked();

    userEvent.click(radio);

    expect(radio).toBeChecked();
  });

  it("matches snapshot", () => {
    const { container } = render(
      <Radio
        id="radio-id"
        className="radio-class"
        label="Test label"
        value="testValue"
        containerAttrs={{ id: "container-id", className: "container-class" }}
      />,
    );

    expect(container).toMatchSnapshot();
  });

  it("matches snapshot with `inline = true`", () => {
    const { container } = render(<Radio id="testId" label="Test label" value="testValue" inline />);

    expect(container).toMatchSnapshot();
  });
});
