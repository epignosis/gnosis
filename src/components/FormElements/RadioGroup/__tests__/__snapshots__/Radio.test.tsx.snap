// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Radio /> matches snapshot 1`] = `
<div>
  <div
    class="container-class css-12utew3-radioButtonContainer-radioButtonContainer"
    id="container-id"
  >
    <input
      class="radio-class"
      id="radio-id"
      type="radio"
      value="testValue"
    />
    <label
      for="radio-id"
    >
      Test label
    </label>
  </div>
</div>
`;

exports[`<Radio /> matches snapshot with \`inline = true\` 1`] = `
<div>
  <div
    class="css-ud70d6-radioButtonContainer-radioButtonContainer"
  >
    <input
      id="testId"
      type="radio"
      value="testValue"
    />
    <label
      for="testId"
    >
      Test label
    </label>
  </div>
</div>
`;
